DEPENDENCY_POLICY.md

```diff
@@ -0,0 +1,91 @@
+# Envoy External Dependency Policy
+
+Envoy has an evolving policy on external dependencies, tracked at
+https://github.com/envoyproxy/envoy/issues/10471. This will become stricter over time, below we
+detail the policy as it currently applies.
+
+## Declaring external dependencies
+
+In general, all external dependencies for the Envoy proxy binary build and test should be declared
+in either [bazel/repository_locations.bzl](bazel/repository_locations.bzl) or
+[api/bazel/repository_locations.bzl](api/bazel/repository_locations.bzl), unless listed under
+[policy exceptions](#policy-exceptions).
+
+An example entry for the `nghttp2` dependency is:
+
+```python
+com_github_nghttp2_nghttp2 = dict(
+    project_name = "Nghttp2",
+    project_url = "https://nghttp2.org",
+    version = "1.41.0",
+    sha256 = "eacc6f0f8543583ecd659faf0a3f906ed03826f1d4157b536b4b385fe47c5bb8",
+    strip_prefix = "nghttp2-{version}",
+    urls = ["https://github.com/nghttp2/nghttp2/releases/download/v{version}/nghttp2-{version}.tar.gz"],
+    use_category = ["dataplane"],
+    cpe = "cpe:2.3:a:nghttp2:nghttp2:*",
+),
+```
+
+Dependency declarations must:
+
+* Provide a meaningful project name and URL.
+* State the version in the `version` field. String interpolation should be used in `strip_prefix`
+  and `urls` to reference the version. If you need to reference version `X.Y.Z` as `X_Y_Z`, this
+  may appear in a string as `{version}.replace('.','_')`.
+* Provide accurate entries for `use_category`. Please think carefully about whether there are data
+  or control plane implications of the dependency.
+* CPEs are compulsory for all dependencies that are not purely build/test.
+  [CPEs](https://en.wikipedia.org/wiki/Common_Platform_Enumeration) provide metadata that allow us
+  to correlate with related CVEs in dashboards and other tooling, and also provide a machine
+  consumable join key. You can consult the latest [CPE
+  dictionary](https://nvd.nist.gov/products/cpe) to find a CPE for a dependency.`"N/A"` should only
+  be used if no CPE for the project is available in the CPE database. CPEs should be _versionless_
+  with a `:*` suffix, since the version can be computed from `version`.
+
```

Comment:
We should we make choosing a developer provided tarball over a Github generated tarball policy IMO - https://github.com/envoyproxy/envoy/blob/master/bazel/EXTERNAL_DEPS.md#choosing-tarballs

Reply:
I assume you mean release tar ball, rather than generated SHA tarball, clarified.